linode/manager
# refactor: [M3-5194] - RQ-ify Types endpoints

## Description 📝

Migrate all Linode Type endpoints to React Query. Types refer to the plans available when configuring a Linode/LKE cluster (i.e., shared, GPU, bare metal, etc.)

The main complication I found during this refactor was the problem of "shadow plans." These are legacy linode types that aren't returned by the main `/types` and `/types-legacy` endpoints but can be fetched individually through `/types/{ID}`.

Originally, this problem was solved by a bit of code `AuthenticationWrapper` that would load all the account linodes on app startup, individually fetch all the unresolved types, and append them to the Redux store.

Now, this is handled through a second RQ hook called `useSpecificTypes()` which takes in an array of type IDs and fetches each one individually. I made sure to use this hook everywhere a "shadow plan" might be present, but I don't have any real examples of a linode with such a type in my account to test on, so I'm not certain all edge cases are handled.

Open Questions:

- [x] Are there still Linodes with "shadow plans" out there in the wild?
- [x] If so, is it possible to configure a new Linode with a such a plan for testing purposes?

## How to test 🧪
1. Ensure all affected components/hooks fetch and display plan details and pricing info as expected:

- `AddNodePoolDrawer`
- `BackupDrawer`
- `CreateCluster`
- `EnableBackupsDialog`
- `KubeCheckoutBar`
- `KubeClusterSpecs`
- `KubernetesClusterRow`
- `LinodeActionMenu`
- `LinodeCreateContainer`
- `LinodeEntityDetail`
- `LinodeResize`
- `LinodeTransferTable`
- `MigrateLinode`
- `MutationNotification`
- `NodePoolsDisplay`
- `RenderProgressEvent`
- `ResizeNodePoolDrawer`
- `SearchBar`
- `SearchLanding`
- `useAPISearch`
- `useExtendedLinode`

2. Unit tests: `yarn test`
3. E2E tests: `yarn cy:e2e`
